src/wstool/multiproject_cli.py

```diff
@@ -621,6 +622,99 @@ def cmd_diff(self, target_path, argv, config=None):
 
         return False
 
+    def cmd_foreach(self, target_path, argv, config=None):
+        """Run shell commands in each repository."""
+        parser = OptionParser(
+            usage='usage: %s foreach [localname]* [command] [OPTIONS]' % self.progname,
+            description=__MULTIPRO_CMD_DICT__['foreach'],
+            epilog='See: http://www.ros.org/wiki/rosinstall for details')
+        parser.add_option('--shell', default=False,
+                          help='use the shell as the program to execute',
+                          action='store_true')
+        parser.add_option('--no-stdout', dest='show_stdout',
+                          default=True,
+                          help='do not show stdout',
+                          action='store_false')
+        parser.add_option('--no-stderr', dest='show_stderr',
+                          default=True,
+                          help='do not show stderr',
+                          action='store_false')
+        parser.add_option("--git", dest="git", default=False,
+                          help="run command in git entries",
+                          action="store_true")
+        parser.add_option("--svn", dest="svn", default=False,
+                          help="run command in svn entries",
+                          action="store_true")
+        parser.add_option("--hg", dest="hg", default=False,
+                          help="run command in hg entries",
+                          action="store_true")
+        parser.add_option("--bzr", dest="bzr", default=False,
+                          help="run command in bzr entries",
+                          action="store_true")
+        # -t option required here for help but used one layer above
+        # see cli_common
+        parser.add_option("-t", "--target-workspace", dest="workspace",
+                          default=None,
+                          help="which workspace to use",
+                          action="store")
+        (options, args) = parser.parse_args(argv)
+
+        if args:
+            localnames, command = args[:-1], args[-1]
+            localnames = localnames if localnames else None
+        else:
+            print("Error: Too many arguments.")
```

Comment:
That error message seems wrong at a glance. Shouldn't it be "Missing arguments"?